tripy/tripy/backend/mlir/utils.py

```diff
@@ -85,6 +85,7 @@ def get_mlir_scalar_attr(mlir_dtype, value):
     # MLIR represents float dtypes as FloatAttr
     # and non-float dtypes as IntegerAttr
     attr_func = ir.IntegerAttr.get if isinstance(mlir_dtype, ir.IntegerType) else ir.FloatAttr.get
+    value = int(value) if isinstance(mlir_dtype, ir.IntegerType) else float(value)
```

Comment:
Do you know why we were getting the wrong type here in the first place?

Reply:
It came from here in `reshape.py`:

```python
def infer_dimensions(input: "tripy.Tensor", shape: ShapeLike) -> ShapeLike:

    num_unknown_dims = len([dim for dim in shape if op_utils.is_minus_one(dim)])
    if num_unknown_dims > 1:
        raise_error(f"The new shape can have at most one inferred dimension (denoted by -1)", [f"Got shape: {shape}."])

    if num_unknown_dims == 1:
        input_volume = math.prod(input.shape)
        known_dims_volume = math.prod(dim for dim in shape if not op_utils.is_minus_one(dim))
        inferred_dim = input_volume / known_dims_volume # <- output is floating point

        shape = [inferred_dim if op_utils.is_minus_one(dim) else dim for dim in shape]

    return {"shape": shape}
```

We don't get a floating point value on that line if we have `DimensionSize` since it would be doing `int32` division, but we do for a Python number.

Reply:
I see. I wonder if we should do a floor division to more accurately convey the intent?

Reply:
It seems floor division for tensors is just division and doesn't do any casting. I will see if it's easy to change it to also include a cast to an int type.

Reply:
Replaced it here and removed the cast in utils. I can put it back if we prefer but it might be handy to have something fail if we get the type wrong.